pyanaconda/storage/osinstall.py

```diff
@@ -230,13 +230,15 @@ def reset(self, cleanup_only=False):
 
         super().reset(cleanup_only=cleanup_only)
 
+        # Protect devices from teardown.
+        self._mark_protected_devices()
+        self.devicetree.teardown_all()
+
         self.fsset = FSSet(self.devicetree)
 
         # Clear out attributes that refer to devices that are no longer in the tree.
         self.bootloader.reset()
 
-        self._mark_protected_devices()
-
```

Comment:
What is the reason to move this code before FSSet and reset of the bootloader?

Reply:
To protect devices from teardown and make the teardown as soon after reset as possible.

Reply:
OK, thanks for the explanation.